modules/internal/ChapelArray.chpl

```diff
@@ -1558,37 +1558,59 @@ module ChapelArray {
     /* Return true if this domain is a subset of ``super``. Otherwise
        returns false. */
     proc isSubset(super : domain) {
-      if !isAssociativeDom(this) {
-        if isRectangularDom(this) then
-          compilerError("isSubset not supported on rectangular domains");
-        else if isOpaqueDom(this) then
+      if !(isAssociativeDom(this) || isRectangularDom(this)) {
+        if isOpaqueDom(this) then
           compilerError("isSubset not supported on opaque domains");
         else if isSparseDom(this) then
           compilerError("isSubset not supported on sparse domains");
         else
           compilerError("isSubset not supported on this domain type");
       }
       if super.type != this.type then
-        compilerError("isSubset called with different associative domain types");
+        if isRectangularDom(this) {
+          if super.dims().size != this.dims().size then
+            compilerError("isSubset called with different rectangular domain types");
+        } else
+          compilerError("isSubset called with different associative domain types");
+
+      if isRectangularDom(this) {
+        var contains = true;
+        for i in 1..this.dims().size {
+          contains &&= super.dims()[i].contains(this.dims()[i]);
+          if contains == false then break;
+        }
```

Comment:
I think this loop is correct, but a simpler implementation might be to just check .first and .last:
```chpl
return super.contains(this.first) && super.contains(this.last);
```

Reply:
Wouldn't that not work right for strided ranges?

Reply:
Wouldn't this create a problem if the object `super` is strided?
For eg, 
```chpl
writeln({1..5}.isSubset({1..5 by 2}));
```
It should output `false` in this case, but if we check only the bounds, it'd output `true`

Reply:
> Wouldn't that not work right for strided ranges?

Oh, missed out this comment.

Reply:
No worries. We'll see if @benharsh agrees.

Reply:
Ah, you're both right, my mistake!